utils/utils.go

```diff
@@ -0,0 +1,62 @@
+package utils
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#package-names
please don't start the same `utils` kitchen-sink package here.
how about `utils/nettest`?

also, https://github.com/golang/go/wiki/CodeReviewComments#package-comments